core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -64,28 +60,23 @@
     private static final double SMALL_DICTIONARIES_MAX_CARDINALITY_RATIO = .25;
 
     private final List<Type> types;
-    private final List<Type> hashTypes;
     private final int[] channels;
 
     private final PagesHashStrategy hashStrategy;
+    // There is always only one builder per block. This List is made to satisfy API of {@link PagesHashStrategy}
     private final List<ObjectArrayList<Block>> channelBuilders;
     private final Optional<Integer> inputHashChannel;
     private final HashGenerator hashGenerator;
     private final OptionalInt precomputedHashChannel;
     private final boolean processDictionary;
-    private PageBuilder currentPageBuilder;
-
-    private long completedPagesMemorySize;
```

Comment:
Would increasing max page size in `PageBuilder` achieve similar results?

Reply:
I don't think so. The gain is IMO the result of the removal of two-level addressing.